### ISSUE TITLE
Error: Failed to obtain extra user information

### ISSUE THREAD

**Issue by soapsurfer** [jgeboski/bitlbee-steam#89]
Hello,
I'm again running into disconnects from time to time. Bitlbee output looks like the following:

steam - Logging in: Logged in
steam - Error: Failed to obtain extra user information
steam - Signing off.. 
steam - Reconnecting in 5 seconds..
steam - Logging in: Connecting
steam - Logging in: Sending logon request
steam - Logging in: Requesting friends list
steam - Logging in: Logged in

bitlbee and bitlbee-steam are up to date.
Any ideas on how to further investigate or fix the issue?

**Comment by jgeboski** [comment]
We have no control over this happening as this is Steam is burping.  Just wait it out, and things should eventually start working.  The plugin is eventually going to stop making this call once #87 is done, which may or may not help.

**Comment by kirbo** [comment]
I'm unable to login into ```bitlbee-steam```. I have:
- Debian GNU/Linux 8.1 (jessie)
- Bitlbee 3.4.1+20150811+master+17-g86fd261-git
- Bitlbee-common 3.4.1+20150811+master+17-g86fd261-git
- Bitlbee-dev 3.4.1+20150811+master+17-g86fd261-git
- Bitlbee-facebook commit hash 60a42307f24c617029efbb7bcd432d162706c692
- Bitlbee-steam commit hash f558aafba97f1f43bdca32d2f86f05ff24630646

```
[13/08 01:39:52] <@Kirbo> account add steam <user> <pass>
[13/08 01:39:52] <@root> Account successfully added with tag steam
[13/08 01:39:54] <@Kirbo> account on
[13/08 01:39:54] <@root> Trying to get all accounts connected...
[13/08 01:39:54] <@root> steam - Logging in: Connecting
[13/08 01:39:54] <@root> steam - Logging in: Requesting authentication key
[13/08 01:39:55] <@root> steam - Logging in: Requesting authentication token
[13/08 01:39:56] <@root> steam - Login error:
[13/08 01:39:56] <@root> steam - Logging in: Run: account steam set authcode <code>
[13/08 01:39:56] <@root> steam - Logging in: Signing off..

[13/08 01:40:24] <@Kirbo> account steam set authcode <code>
[13/08 01:40:24] <@root> steam - Logging in: Connecting
[13/08 01:40:24] <@root> steam - Logging in: Requesting authentication key
[13/08 01:40:24] <@root> authcode = `<code>'
[13/08 01:40:24] <@root> steam - Logging in: Requesting authentication token
[13/08 01:40:26] <@root> steam - Logging in: Authentication finished
[13/08 01:40:26] <@root> steam - Logging in: Signing off..
[13/08 01:40:26] <@root> steam - Logging in: Connecting
[13/08 01:40:26] <@root> steam - Logging in: Sending logon request
[13/08 01:40:30] <@root> steam - Login error: Failed to obtain extra user information
[13/08 01:40:30] <@root> steam - Logging in: Signing off..
[13/08 01:40:30] <@root> steam - Logging in: Reconnecting in 5 seconds..
[13/08 01:40:35] <@root> steam - Logging in: Connecting
[13/08 01:40:35] <@root> steam - Logging in: Sending logon request
[13/08 01:40:37] <@root> steam - Login error: Failed to obtain extra user information
[13/08 01:40:37] <@root> steam - Logging in: Signing off..
[13/08 01:40:37] <@root> steam - Logging in: Reconnecting in 15 seconds..
...and so on
```

And debug data:
```
About to send HTTP request:
POST /ISteamWebUserPresenceOAuth/Logon/v0001 HTTP/1.1
User-Agent: Steam App / BitlBee / 1.2.0
Content-Length: 74
Connection: Close
Accept: */*
Cookie: steamLogin=0%7C%7Coauth%3Acc00c718313b0ec08d8f77695a8a297d; sessionid=d150b5f00a9568c42e7c6eec
Host: api.steampowered.com
Content-Type: application/x-www-form-urlencoded

ui_mode=web&access_token=cc00c718313b0ec08d8f77695a8a297d&umqid=2551433647
HTTP response headers:
HTTP/1.1 200 OK
Date: Wed, 12 Aug 2015 22:49:43 GMT
expires: Wed, 12 Aug 2015 22:49:43 GMT
Content-Type: application/json; charset=UTF-8
content-length: 158

Finishing HTTP request with status: 200 OK
About to send HTTP request:
GET /ISteamUserOAuth/GetUserSummaries/v0001?steamids=76561198014256834&access_token=cc00c718313b0ec08d8f77695a8a297d HTTP/1.1
User-Agent: Steam App / BitlBee / 1.2.0
Connection: Close
Accept: */*
Cookie: steamLogin=76561198014256834%7C%7Coauth%3Acc00c718313b0ec08d8f77695a8a297d; sessionid=d150b5f00a9568c42e7c6eec
Host: api.steampowered.com


HTTP response headers:
HTTP/1.1 200 OK
Date: Wed, 12 Aug 2015 22:49:44 GMT
expires: Wed, 12 Aug 2015 22:49:44 GMT
Content-Type: application/json; charset=UTF-8
content-length: 777

Finishing HTTP request with status: 200 OK
About to send HTTP request:
GET /chat/? HTTP/1.1
User-Agent: Steam App / BitlBee / 1.2.0
Connection: Close
Accept: */*
Cookie: steamLogin=76561198014256834%7C%7Coauth%3Acc00c718313b0ec08d8f77695a8a297d; sessionid=d150b5f00a9568c42e7c6eec
Host: steamcommunity.com


HTTP response headers:
HTTP/1.1 302 Moved Temporarily
Server: Apache
X-Frame-Options: DENY
Content-Security-Policy: script-src 'self' 'unsafe-inline' 'unsafe-eval' https://steamcommunity-a.akamaihd.net/ https://api.steampowered.com/ http://www.google-analytics.com https://ssl.google-analytics.com https://www.google.com https://www.gstatic.com https://apis.google.com; object-src 'none'; connect-src 'self' http://steamcommunity.com https://steamcommunity.com https://api.steampowered.com/; frame-src 'self' http://store.steampowered.com/ https://store.steampowered.com/ http://www.youtube.com https://www.youtube.com https://www.google.com;
Expires: Mon, 26 Jul 1997 05:00:00 GMT
Cache-Control: no-cache
Location: https://steamcommunity.com/login/home/?goto=%2Fchat%2F%3F
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 20
Content-Type: text/html; charset=UTF-8
Date: Wed, 12 Aug 2015 22:49:45 GMT
Connection: close
Set-Cookie: steamCountry=FI%7Cd552c34021e392c7185ac8258445e75c; path=/

Finishing HTTP request with status: 302 Moved Temporarily

```

What more can I provide and/or what should I do?

**Comment by jgeboski** [comment]
This is due to http://steamcommunity.com/chat/ returning a gzipped response, which bitlbee's HTTP client does not handle.  I have a patch in the works now to remove this request (#87).

**Comment by kirbo** [comment]
Thanks for the quick reply :+1: 
Okay, so there is no workaround for it at the time? All I can do is just wait for the patch?

**Comment by jgeboski** [comment]
@kirbo You can disable the fetching of additional information with this [patch](https://gist.github.com/jgeboski/41b9386219f72802e747).

**Comment by kirbo** [comment]
That seemed to be working, thanks :) So you're working on something bigger/better patch/commit for this gzipped response issue, or was it the patch you linked?

**Comment by jgeboski** [comment]
Something bigger and better, #87.

**Comment by gdude2002** [comment]
Just gonna throw in my upvote; I've been getting this error persistently since last night (having used bitlbee-steam successfully for quite some time - the repeated error showed up while I was using it), and it's been impossible to sign into Steam using bitlbee since then.

**Comment by jgeboski** [comment]
This should be resolved as of aaf2cbc.

**Comment by kirbo** [comment]
Works like a charm, thanks again :+1:
